docs/root/intro/arch_overview/http/http_proxy.rst

```diff
@@ -40,15 +39,17 @@ For further configuration information see the :ref:`HTTP filter configuration do
 Memory usage details
 --------------------
 
-.. attention::
-
-  HTTP dynamic forward proxy support currently can use unbounded memory and is not ready for
-  production use. Circuit breakers and other limits will be added soon.
-
 Memory usage detail's for Envoy's dynamic forward proxy support are as follows:
 
 * Each resolved host/port pair uses a fixed amount of memory global to the server and shared
   amongst all workers.
 * Address changes are performed inline using read/write locks and require no host reallocations.
 * Hosts removed via TTL are purged once all active connections stop referring to them and all used
   memory is regained.
+* The :ref:`max_hosts
+  <envoy_api_field_config.common.dynamic_forward_proxy.v2alpha.DnsCacheConfig.max_hosts>` field can
+  be used to limit the number of hosts that the DNS cache will store at any given time.
+* The cluster's :ref:`max_pending_requests
```

Comment:
Also I do wonder how we're going to want to adapt this to multi-tenancy.  I think offhand we're going to want to have a forward proxy cluster per tenant, but naively I'd think we'd want to share a DNS cache, which I think won't work well with these APIs.  I don't think we need to design this today but can we make sure to have a plan before we move APIs out of alpha?

Reply:
The DNS caches are effectively global and referenced by name, so I think this should work fine?

Reply:
You think?  I'd think we could fairly easily have a dns cache per tenant, and it'd all just work (apart from scaling concerns) but if we wanted a shared cache we'd want separate circuit breaking for the size of an individual cluster and the size of the global dns cache, no?

Reply:
Circuit breakers are per-cluster though?